docs/core/testing/unit-testing-mstest-migration-from-v1-to-v3.md

```diff
@@ -0,0 +1,224 @@
+---
+title: MSTest migration to v3
+description: Migration guide.
+author: engyebrahim
+ms.author: enjieid
+ms.date: 11/06/2024
+---
+
+# MSTest v3 Migration Guide
+
+## Overview
+
+This guide assists users in upgrading their MSTest projects from v0/v1 to MSTest v3. MSTest v3 introduces new features, improvements, and some breaking changes.
+
+---
+
+## Introduction and Motivation
+
+Migrating to MSTest v3 offers several benefits:
+
+- Improved test execution speeds and memory efficiency.
+- Stricter validation for more reliable tests.
+- Leveraging modern .NET features and cross-platform compatibility.
+
+---
+
+## Steps to Migrate
+
+### 1. Remove Assembly Reference
+
+Remove the `Microsoft.VisualStudio.QualityTools.UnitTestFramework` reference from your project.
+
+```xml
+<Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+    <Private>False</Private>
+</Reference>
+```
+
+### 2. Update Your Project
+
+- **NuGet Package**: Install the latest [MSTest](https://www.nuget.org/packages/MSTest).
+- **Project File**: Update your project file to use MSTest SDK.
+
+```xml
+<Project Sdk="MSTest.Sdk/3.3.1">
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+  </PropertyGroup>
+</Project>
+```
+
+### 3. Update Your Code
+
+- **Replace Deprecated Methods**: Update deprecated methods to newer versions.
```

Comment:
Are the deprecated methods just deprecated, or deprecated with "break build"? It would be nice to point add the the previous point that you should be able to build,  with deprecated warnings.

Reply:
it's removed

Reply:
If the functionality is removed, then please don't say deprecated, but maybe replaced? Deprecated in api is usually meaning that the functionality is still there but is not the modern way to do stuff and will be removed later.